### DISCUSSION TITLE
Payment gateway

### DISCUSSION THREAD

**Discussion by demodes** [ecomplus/storefront#497] [Q&A]
I was not able to find a payment gateway config. How can I set up a payment for the shop? Where can I found the settings to set up the payment gateway config? 
I don't know where I should ask these questions. I have not access to the forum.

**Comment by leomp12** [comment]
Hey @denisdemolli ,

Which gateway do you want to integrate?

You can simply "mock" payment gateways client side only (I can give an example to you) to handle payment buttons such as Stripe.

Or with an [E-Com Plus](https://www.e-com.plus/) merchant account (we can provide it free for dev or a trial for stores in some countries) you can start from https://github.com/ecomplus/application-starter to setup full featured payment gateway apps and install in your store, such as:

https://github.com/ecomplus/app-pagarme
https://github.com/ecomplus/app-mercadopago
https://github.com/ecomplus/app-paypal
...

> I don't know where I should ask these questions. I have not access to the forum.

Currently, forum is open for [E-Com Plus](https://www.e-com.plus/) merchants and partners, we usually don't have technical discussions there either. Best place to ask this would be at repo [discussions](https://github.com/ecomplus/storefront/discussions).

**Comment by demodes** [comment]
Hi Leonardo, 

thank you for the response.

if I understand it correctly It means I can not take this code and add backend payment functionality right now. I would need access to other proprietary software.

I would completely understand that btw. I just want to better understand the structure of the project.

If you don't mind can you give me an example or some general advice on how to integrate the stripe(client-side) button into the storefront?

**Reply by leomp12** [reply]
@denisdemolli sure you can, we just haven't docs for it yet 😬 

As our Storefront is a Jamstack app, to add any backend functionality you must have a component (eg.: a payment button) that call an API endpoint (eg.: a middleware to check params and create transaction on your gateway).

Give me some time, I'll give you an example hard setting payment method client side and calling an endpoint on custom payment button click.

**Reply by leomp12** [reply]
Stripe is just a way to setup it easier (worrying only about payment button at the frontend) without using our APIs.
